openlibrary/plugins/importapi/code.py

```diff
@@ -128,17 +135,17 @@ def POST(self):
         except ClientException as e:
             return self.error('bad-request', **json.loads(e.json))
 
-    def reject_non_book_marc(self, marc_record, **kwargs):
-        details = 'Item rejected'
-        # Is the item a serial instead of a book?
-        marc_leaders = marc_record.leader()
-        if marc_leaders[7] == 's':
-            return self.error('item-is-serial', details, **kwargs)
+def raise_non_book_marc(marc_record, **kwargs):
+    details = 'Item rejected'
+    # Is the item a serial instead of a book?
+    marc_leaders = marc_record.leader()
+    if marc_leaders[7] == 's':
+        raise BookImportError(('item-is-serial', details, kwargs))
 
-        # insider note: follows Archive.org's approach of
-        # Item::isMARCXMLforMonograph() which excludes non-books
-        if not (marc_leaders[7] == 'm' and marc_leaders[6] == 'a'):
-            return self.error('item-not-book', details, **kwargs)
+    # insider note: follows Archive.org's approach of
+    # Item::isMARCXMLforMonograph() which excludes non-books
+    if not (marc_leaders[7] == 'm' and marc_leaders[6] == 'a'):
+        raise BookImportError(('item-not-book', details, kwargs))
```

Comment:
Do these kwargs get passed through to the final API response? There might be a better way to handle the bulk MARC response now that the method is split.

Reply:
I believe so

Reply:
Given the bulk MARC code didn't touch this path and I don't know the internals very well, I'm leaving this for a future PR